native/src/hooks/__tests__/useNavigateToDeepLink.spec.tsx

```diff
@@ -0,0 +1,411 @@
+import { waitFor } from '@testing-library/react-native'
+import { mocked } from 'jest-mock'
+import React, { useEffect } from 'react'
+
+import {
+  CATEGORIES_ROUTE,
+  EVENTS_ROUTE,
+  INTRO_ROUTE,
+  JPAL_TRACKING_ROUTE,
+  LANDING_ROUTE,
+  LOCAL_NEWS_TYPE,
+  NEWS_ROUTE,
+  OFFERS_ROUTE,
+  OPEN_DEEP_LINK_SIGNAL_NAME,
+} from 'api-client'
+import { FeatureFlagsType } from 'build-configs/BuildConfigType'
+
+import buildConfig from '../../constants/buildConfig'
+import { AppContext } from '../../contexts/AppContextProvider'
+import createNavigationPropMock from '../../testing/createNavigationPropMock'
+import render from '../../testing/render'
+import appSettings from '../../utils/AppSettings'
+import sendTrackingSignal from '../../utils/sendTrackingSignal'
+import useNavigate from '../useNavigate'
+import useNavigateToDeepLink from '../useNavigateToDeepLink'
+import useSnackbar from '../useSnackbar'
+
+jest.mock('../useNavigate')
+jest.mock('../useSnackbar')
+jest.mock('../../utils/sendTrackingSignal')
+
+describe('useNavigateToDeepLink', () => {
+  const mockedBuildConfig = mocked(buildConfig)
+  const showSnackbar = jest.fn()
+  const navigateTo = jest.fn()
+  const navigation = createNavigationPropMock()
+  mocked(useSnackbar).mockImplementation(() => showSnackbar)
+  mocked(useNavigate).mockImplementation(() => ({ navigateTo, navigation }))
+
+  const changeCityCode = jest.fn()
+  const changeLanguageCode = jest.fn()
+  const selectedLanguageCode = 'de'
+
+  const mockBuildConfig = (featureFlags: Partial<FeatureFlagsType>) => {
+    const previous = buildConfig()
+    // @ts-expect-error partial of fixed city type leads to problems
```

Comment:
Basically we have following type as part of the feature flags of the build config:
``` ts
export type FixedCityType =
  | {
      // Shows intro slides to the users on first app start.
      introSlides: false
      // Preselects a city without showing a selection, changing it is not possible for users.
      fixedCity: string | null
    }
  | {
      introSlides: true
      fixedCity: null
    }
```
TS is complaining because we are passing only a partial of that type, so it may be possible to pass `fixedCity: `augsburg` while `introSlides: true`. But for testing this is not a problem and not worth another fix.

Reply:
Adjusted comment

Reply:
Thanks, got it now :)